app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,84 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :set_posts, only: [:index]
+  # GET /posts
+  # GET /posts.json
+  def index
+
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show
+    @posts=Post.find(params[:id])
+    @posts=[@posts]
+  end
+
+  # GET /posts/new
+  def new
+    @post = Post.new
+  end
+
+  # GET /posts/1/edit
+  def edit
+  end
+
+  # POST /posts
+  # POST /posts.json
+
+  def create
+    @post = Post.new(post_params) do |post|
+      post.user = current_user
+      post.attachment=""
+    end
```

Comment:
Create a method in  `Post` model and Move this logic there.

Reply:
Done